### ISSUE TITLE
Ability to update span after it is created

### ISSUE THREAD

**Issue by honghzzhang** [openzipkin/zipkin-go#51]
In the current zipkin-go project, there are various SpanOptions functions (for example, func Kind(kind model.Kind) SpanOption)) to adjust a span that is being created. However, there is no API to adjust a span that is already created (please let me know if I missed it). 

In brave API,  you can update Span after it is created, for example:
Span kind(Span.Kind var1) which sets the kind of an existing span
Span name(String var1) which sets the name of an existing span

Can we add this functionality to zipkin-go as well?

**Comment by basvanbeek** [comment]
As also mentioned in #50 we have deliberately kept the API surface small and have refrained from adding span modification methods as much as possible. In Go I have not not seen the need for post creation updates to information like span Kind as typically this is known at span creation time in Go.

Span name typically is something that might need to be updated after span creation if using particular frameworks or toolkits. 

Go kit for example needs to start spans before the serviced Go kit endpoint name is known. So for this real use case we do have an API method: https://github.com/openzipkin/zipkin-go/blob/master/span.go#L15

Another example of items that might only become available after span creation is RemoteEndpoint. So we have a method for that too.

Long story short. We only allow span modification methods into the API if a particular framework or toolkit we want to support can't provide those details at span creation time. Modifying span details after creation comes with additional cost which most people are not very aware of.

**Comment by adriancole** [comment]
I want to just drop in and mention that zipkin-go is not a brave port,
rather it is an idiomatic api based on the same v2 data model. There is no
imperative for it to behave exactly like brave.. in fact brave has some
design decisions just because of its history (ex supporting migration off
brave v3)

Anyway just wanted to reiterate that best decisions trump consistency.
Where there is good reason for things to be different they should. These
are different libraries and between bas, jose carlos and future
contributors (maybe even you!).. decisions on this library is best placed
in the hands of those maintaining it.

**Comment by honghzzhang** [comment]
@basvanbeek: Thanks for explaining the thinking behind the Span APIs. 
@adriancole: Point taken.
As I mentioned in the other issue, I am currently working on converting a Java project which uses brave API for tracing to a Go project. As an initial pass, I am go through all the used brave APIs, and try to find a mapping in the  zipkin-go project. For the ones I could not find, I would like to understand which case it is with your guys' help: 
1. The API functionality is there, but achieved through a different mechanism.
2. The API is not applicable in Go, so it's not there by design.
3. There is a valid use case for the API and makes sense to add the API. 

Looks like it is case 2) for this particular API. I will try to re-organize/re-write that part of the project to avoid the need of setting span kind after the span is created.

**Comment by honghzzhang** [comment]
Mark this issue as resolved/closed.

**Comment by adriancole** [comment]
Thanks for letting us know what you are up to!
